pkg/dra/mapper.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+
+	configapi "sigs.k8s.io/kueue/apis/config/v1beta1"
+)
+
+var (
+	globalMapper     *ResourceMapper
+	globalMapperOnce sync.Once
+)
+
+// ResourceMapper provides device class to logical resource name mapping
```

Comment:
https://github.com/kubernetes-sigs/kueue/pull/5873/files#r2349039588

Should we check that deviceClasses actually exist?